CONTRIBUTING.md

```diff
@@ -0,0 +1,324 @@
+        self.init_weights()
+
+    def forward(
+        self,
+        token_ids: torch.Tensor,
+        time_delay_ids: torch.Tensor,
+        structure_ids: torch.Tensor,
+        token_attention_mask=None,
+        post_attention_mask=None,
+        labels: Optional[torch.Tensor] = None,
+    ):
+        X = self.wordEncoder(token_ids)
+        word_pos = self.prepare_word_pos(token_ids).to(X.device)
+        word_pos = self.positionEncoderWord(word_pos)
+        time_delay = self.positionEncoderTime(time_delay_ids)
+
+        logits = self.hierarchicalTransformer(
+            X,
+            word_pos,
+            time_delay,
+            structure_ids,
+            attention_mask_word=token_attention_mask,
+            attention_mask_post=post_attention_mask,
+        )
+
+        if labels is not None:
+            loss = self.loss(logits, labels)
+        else:
+            loss = None
+
+        return RumourDetectionTwitterModelOutput(loss=loss, logits=logits)
+```
+
+There are 3 key things to note in the above implementation.
+
+1. When initialising the model, it is important to invoke the `init_weights()` method. Note the lack of an underscore at the start of the method name. This is required so that the model weights are initialized using the methods defined in the `__init__` method specified in `NewModelPreTrainedModel`.
+
+2. The `forward` method takes in an optional `labels` argument. If this argument is passed to the model, the `forward` method should also return the value of the loss function for that batch of inputs.
+
+3. The `forward` method's output is an object of the `RumourDetectionTwitterModelOutput` dataclass. This dataclass is illustrated in the code snippet below.
+
+```
+from dataclasses import dataclass
+from transformers.file_utils import ModelOutput
+
+
+@dataclass
+class RumourDetectionTwitterModelOutput(ModelOutput):
+    """
+    Base class for outputs of Rumour Detection models
+    Args:
+        loss (:obj:`torch.Tensor` of shape `(1,)`, `optional`, returned when :obj:`labels` is provided):
+            Classification loss, typically cross entropy. Loss function used is dependent on what is specified in RumourDetectionTwitterConfig.
+        logits (:obj:`torch.Tensor` of shape :obj:`(batch_size, num_classes)`):
+            Raw logits for each class. num_classes = 4 by default.
+    """
+
+    loss: Optional[torch.Tensor] = None
+    logits: torch.Tensor = None
+```
+
+### train
+
+`train.py` should contain a working implementation of the model training process. A user should be able to train the model from the command line using `python -m train --train_args config/config.json`.
+
+### eval
+
+`eval.py` should contain a working implementing of the model evaluation process. This script should load the trained model (using the information in `config/config.json`) and evaluate it against the evaluation datasets. The evaluation metrics reported should correspond to that reported in the `README.md`.
```

Comment:
Think might be good to mention that there should be `train()` and `evaluate()` functions in train.py and eval.py respectively